### PR TITLE
feat(paddle backend): removed manual dtype casting -- round 2

### PR DESCRIPTION
**Pull request by Madjid-CH**

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

File: ivy/functional/backends/paddle/general.py [hunk 1]
```diff
@@ -45,24 +47,17 @@ def _check_query(query):
     )
 
 
+@with_unsupported_dtypes(
+    {"2.5.1 and below": ("float16", "int16", "int8")}, backend_version
```
Comment (vedpatwardhan) [inline ivy/functional/backends/paddle/general.py:51]: have you tested if the function works with `complex` and `bfloat16`?

File: ivy/functional/backends/paddle/linear_algebra.py [hunk 2]
```diff
@@ -536,6 +525,9 @@ def svdvals(
     return paddle_backend.svd(x)[1]
 
 
+@with_supported_dtypes(
+    {"2.5.1 and below": ("complex", "float32", "float64")}, backend_version
```
Comment (vedpatwardhan) [inline ivy/functional/backends/paddle/linear_algebra.py:529]: are `complex` dtypes supported?

File: ivy/functional/backends/paddle/manipulation.py [hunk 3]
```diff
@@ -69,6 +73,10 @@ def expand_dims(
     return x.reshape(out_shape)
 
 
+@with_unsupported_dtypes(
+    {"2.5.1 and below": ("bfloat16", "float16", "int16", "int8", "uint8")},
```
Comment (vedpatwardhan) [inline ivy/functional/backends/paddle/manipulation.py:77]: I assume you've already verified that `bfloat16` isn't supported for this function? Same applies to the other functions where `bfloat16 ` was marked as unsupported

File: ivy/functional/backends/paddle/manipulation.py [hunk 4]
```diff
@@ -151,6 +158,10 @@ def reshape(
     return ret
 
 
+@with_supported_dtypes(
+    {"2.5.1 and below": ("complex", "float32", "float64", "int32", "int64")},
```
Comment (vedpatwardhan) [inline ivy/functional/backends/paddle/manipulation.py:162]: have you tested `complex` dtypes? Same for other functions in this file where `complex` has been marked as supported

File: ivy/functional/backends/paddle/manipulation.py [hunk 5]
```diff
@@ -378,11 +374,21 @@ def tile(
...
 
 
+@with_unsupported_dtypes(
+    {
+        "2.5.1 and below": (
+            "bfloat16",
+            "float16",
+            "int8",
+            "int16",
+            "uint16",
```
Comment (vedpatwardhan) [inline ivy/functional/backends/paddle/manipulation.py:387]: ig this should be `uint8` instead of `uint16`

**Review by vedpatwardhan — CHANGES_REQUESTED** [review]
Just requested a few changes, thanks @Madjid-CH 😄

File: ivy/functional/backends/paddle/manipulation.py (hunk 3 above)
Comment (Madjid-CH) [reply ivy/functional/backends/paddle/manipulation.py:77]: yes I verified `bfloat16` here and it's not supported.
my approach for all functions modified here is to loop through every `dtype `in `paddle ` and call the function with this `dtype` and if there's not error or strange behavior( like input dtype not the same as output dtype) then I'll mark the dtype as supported.
Hope that make sense 😀

File: ivy/functional/backends/paddle/manipulation.py (hunk 4 above)
Comment (Madjid-CH) [reply ivy/functional/backends/paddle/manipulation.py:162]: yep:
```python
import paddle
from ivy.functional.backends.paddle.manipulation import flip
import ivy
ivy.set_paddle_backend()
for dtype in ["complex64", "complex128"]:
    x = paddle.to_tensor(
        [
                [1, 2],
        ],
        dtype=dtype,
    )
    print(dtype, flip(x))

complex64 Tensor(shape=[1, 2], dtype=complex64, place=Place(cpu), stop_gradient=True,
       [[(2+0j), (1+0j)]])
complex128 Tensor(shape=[1, 2], dtype=complex128, place=Place(cpu), stop_gradient=True,
       [[(2+0j), (1+0j)]])
```

Comment (Madjid-CH) [reply ivy/functional/backends/paddle/manipulation.py:162]:
```python
import paddle
from ivy.functional.backends.paddle.manipulation import roll as fn
import ivy
ivy.set_paddle_backend()
for dtype in ["complex64", "complex128"]:
    x = paddle.to_tensor(
        [
                [1, 2],
        ],
        dtype=dtype,
    )
    print(dtype, fn(x, 1))
           

complex64 Tensor(shape=[1, 2], dtype=complex64, place=Place(cpu), stop_gradient=True,
       [[(2+0j), (1+0j)]])
complex128 Tensor(shape=[1, 2], dtype=complex128, place=Place(cpu), stop_gradient=True,
       [[(2+0j), (1+0j)]])
```

Comment (Madjid-CH) [reply ivy/functional/backends/paddle/manipulation.py:162]:
```python
import paddle
from ivy.functional.backends.paddle.manipulation import permute_dims as fn
import ivy
ivy.set_paddle_backend()
for dtype in ["complex64", "complex128"]:
    x = paddle.to_tensor(
        [
                [1, 2],
        ],
        dtype=dtype,
    )
    print(dtype, fn(x, axes=(1, 0)))

complex64 Tensor(shape=[2, 1], dtype=complex64, place=Place(cpu), stop_gradient=True,
       [[(1+0j)],
        [(2+0j)]])
complex128 Tensor(shape=[2, 1], dtype=complex128, place=Place(cpu), stop_gradient=True,
       [[(1+0j)],
        [(2+0j)]])
```

Comment (Madjid-CH) [reply ivy/functional/backends/paddle/manipulation.py:162]:
```python
import paddle
from ivy.functional.backends.paddle.manipulation import tile as fn
import ivy
ivy.set_paddle_backend()
for dtype in ["complex64", "complex128"]:
    x = paddle.to_tensor(
        [
                [1, 2],
        ],
        dtype=dtype,
    )
    print(dtype, fn(x, repeats=2))
           

complex64 Tensor(shape=[1, 4], dtype=complex64, place=Place(cpu), stop_gradient=True,
       [[(1+0j), (2+0j), (1+0j), (2+0j)]])
complex128 Tensor(shape=[1, 4], dtype=complex128, place=Place(cpu), stop_gradient=True,
       [[(1+0j), (2+0j), (1+0j), (2+0j)]])
```

File: ivy/functional/backends/paddle/manipulation.py (hunk 5 above)
Comment (Madjid-CH) [reply ivy/functional/backends/paddle/manipulation.py:387]: yep, It's a typo error. A sharp reviewer as usual 🚀

File: ivy/functional/backends/paddle/general.py (hunk 1 above)
Comment (vedpatwardhan) [reply ivy/functional/backends/paddle/general.py:51]: awesome, thanks for clarifying!

File: ivy/functional/backends/paddle/linear_algebra.py (hunk 2 above)
Comment (vedpatwardhan) [reply ivy/functional/backends/paddle/linear_algebra.py:529]: awesome, thanks for clarifying!

File: ivy/functional/backends/paddle/manipulation.py (hunk 3 above)
Comment (vedpatwardhan) [reply ivy/functional/backends/paddle/manipulation.py:77]: Yep makes perfect sense, thanks!

File: ivy/functional/backends/paddle/manipulation.py (hunk 4 above)
Comment (vedpatwardhan) [reply ivy/functional/backends/paddle/manipulation.py:162]: awesome, thanks for clarifying!

File: ivy/functional/backends/paddle/manipulation.py (hunk 5 above)
Comment (vedpatwardhan) [reply ivy/functional/backends/paddle/manipulation.py]: thanks for fixing!

**Review by vedpatwardhan — APPROVED** [review]
lgtm! Feel free to merge, thanks @Madjid-CH 😄

**Pull request merged**